lib/src/sfu_connection.dart

```diff
@@ -31,17 +33,29 @@ class SfuConnection {
 
     if (_localStream == null) {
       // Add a transceiver for receiving audio.
-      await _peerConnection.addTransceiver(
+      _audio = await _peerConnection.addTransceiver(
           kind: RTCRtpMediaType.RTCRtpMediaTypeAudio,
           init: RTCRtpTransceiverInit(direction: TransceiverDirection.RecvOnly));
     } else {
       // Add transceivers for sending audio and video.
-      for (MediaStreamTrack track in _localStream!.getTracks()) {
-        // REVIEW: Do we need to set any encoding options, similar to
-        // https://github.com/flutter-webrtc/flutter-webrtc-demo/blob/b2a495d8888fa84da9c8eba164cb2c8d46988a44/lib/src/call_sample/signaling.dart#L352-L373?
-        await _peerConnection.addTransceiver(
-            track: track,
-            init: RTCRtpTransceiverInit(direction: TransceiverDirection.SendOnly, streams: [_localStream!]));
+      // REVIEW: Do we need to set any encoding options, similar to
+      // https://github.com/flutter-webrtc/flutter-webrtc-demo/blob/b2a495d8888fa84da9c8eba164cb2c8d46988a44/lib/src/call_sample/signaling.dart#L352-L373?
+      _audio = await _peerConnection.addTransceiver(
+        init: RTCRtpTransceiverInit(direction: TransceiverDirection.SendOnly, streams: [_localStream!]),
+        kind: RTCRtpMediaType.RTCRtpMediaTypeAudio,
+        track: _localStream!.getAudioTracks()[0],
+      );
+      if (_localStream!.getVideoTracks().isNotEmpty) {
+        _video = await _peerConnection.addTransceiver(
+          init: RTCRtpTransceiverInit(direction: TransceiverDirection.SendOnly, streams: [_localStream!]),
+          kind: RTCRtpMediaType.RTCRtpMediaTypeVideo,
+          track: _localStream!.getVideoTracks()[0],
+        );
+      } else {
+        _video = await _peerConnection.addTransceiver(
+          init: RTCRtpTransceiverInit(direction: TransceiverDirection.SendOnly),
+          kind: RTCRtpMediaType.RTCRtpMediaTypeVideo,
+        );
```

Comment:
An empty send-only video transceiver.

Reply:
I'm actually surprised that the call to addTransceiver doesn't fail, but I like that for its simplicity. Let's go in that direction then.

Reply:
@ipainchaud the idea was borrowed from https://www.w3.org/TR/webrtc/#advanced-peer-to-peer-example-with-warm-up, which I thought I mentioned to you but perhaps forgot. 🤦🏾 

https://microsoft.github.io/MixedReality-WebRTC/manual/glossary.html also references something similar:

> A transceiver always has exactly one [local media track](https://microsoft.github.io/MixedReality-WebRTC/manual/glossary.html#local-media-track) (sender) and one [remote media track](https://microsoft.github.io/MixedReality-WebRTC/manual/glossary.html#remote-media-track) (receiver), both of which can be null. If null, the transceiver acts as if a dummy track existed which draws its media frames from a null source (black frames for video, silence for audio).